### ISSUE TITLE
Channel shows 0 confirmations until it hits 3

### ISSUE THREAD

**Issue by tomsmith8** [stakwork/sphinx-swarm#182]
**Issue**

We are seeing a channel with 1-2 confirmations be shown as having 0 on the UI - until it reaches the third confirmation - that's with CLN as the backend node.

3 confirmation is when CLN marked the channel as "ready for use" and then the respective local remote.

**Comment by fenixlibertas** [comment]
I was asked to take a look at this ticket so I would appreciate additional info..
Can someone please help with the following since I am not familiar with this software:
1. Is there any video explaining the purpose of this software, what are the main features, and how to use it? 
2.  Is there a Software Installation Guide document?
3. Is it possible to run and reproduce this on a local machine, and what steps are required?

**Comment by Evanfeenstra** [comment]
> I was asked to take a look at this ticket so I would appreciate additional info.. Can someone please help with the following since I am not familiar with this software:
> 
> 1. Is there any video explaining the purpose of this software, what are the main features, and how to use it?
> 2. Is there a Software Installation Guide document?
> 3. Is it possible to run and reproduce this on a local machine, and what steps are required?
> 
> Feel free to contact me directly https://discordapp.com/users/691003870482137128

Yes you can run it locally. You will need Docker and Rust. See the README here https://github.com/stakwork/sphinx-swarm/tree/master/src/bin/cln

This project runs bitcoin and lightning nodes, as well as many other services. The `cln` bin is just a test setup that runs 1 bitcoind and 2 CLN nodes so you can test channels.

**Comment by fenixlibertas** [comment]
> > I was asked to take a look at this ticket so I would appreciate additional info.. Can someone please help with the following since I am not familiar with this software:
> > 
> > 1. Is there any video explaining the purpose of this software, what are the main features, and how to use it?
> > 2. Is there a Software Installation Guide document?
> > 3. Is it possible to run and reproduce this on a local machine, and what steps are required?
> > 
> > Feel free to contact me directly https://discordapp.com/users/691003870482137128
> 
> Yes you can run it locally. You will need Docker and Rust. See the README here https://github.com/stakwork/sphinx-swarm/tree/master/src/bin/cln
> 
> This project runs bitcoin and lightning nodes, as well as many other services. The `cln` bin is just a test setup that runs 1 bitcoind and 2 CLN nodes so you can test channels.

Which Operating systems are supported? Ubuntu, MacOS, Windows ?

**Comment by Evanfeenstra** [comment]
any operating system

**Comment by fenixlibertas** [comment]
> any operating system

It seems some dependencies do not support Windows OS.. I'll try to switch to Ubuntu.

PS D:\dev\workspace\sphinx-swarm\src\bin\cln> cargo run --bin cln  
   Compiling cln-grpc v0.1.2 (https://github.com/stakwork/lightning?branch=cln-grpc-serde#0faeef6e)
   Compiling cln-rpc v0.1.2 (https://github.com/stakwork/lightning?branch=cln-grpc-serde#0faeef6e)
   Compiling ubyte v0.10.3
   Compiling uuid v1.4.1
   Compiling bitcoin_hashes v0.9.7
   Compiling unsafe-libyaml v0.2.7
   Compiling ipnet v2.5.0
   Compiling binascii v0.1.4
error: failed to run custom build command for `cln-grpc v0.1.2 (https://github.com/stakwork/lightning?branch=cln-grpc-serde#0faeef6e)`                                             

Caused by:
  process didn't exit successfully: `D:\dev\workspace\sphinx-swarm\target\debug\build\cln-grpc-7ee2d57debca5721\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-changed=proto/node.proto
  cargo:rerun-if-changed=proto

  --- stderr
  thread 'main' panicked at C:\Users\fenix\.cargo\registry\src\index.crates.io-6f17d22bba15001f\prost-build-0.11.4\src\lib.rs:1296:10:
  Could not find `protoc` installation and this build crate cannot proceed without
      this knowledge. If `protoc` is installed and this crate had trouble finding
      it, you can set the `PROTOC` environment variable with the specific path to your
      installed `protoc` binary.You can download it from https://github.com/protocolbuffers/protobuf/releases or from your package manager.

  For more information: https://docs.rs/prost-build/#sourcing-protoc

  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error[E0432]: unresolved import `tokio::net::unix`                                                                                                                                 
  --> C:\Users\fenix\.cargo\git\checkouts\lightning-21c71d1e1901e9d2\0faeef6\cln-rpc\src\lib.rs:12:17
   |
12 | use tokio::net::unix::{OwnedReadHalf, OwnedWriteHalf};
   |                 ^^^^ could not find `unix` in `net`

error[E0432]: unresolved import `tokio::net::UnixStream`
  --> C:\Users\fenix\.cargo\git\checkouts\lightning-21c71d1e1901e9d2\0faeef6\cln-rpc\src\lib.rs:13:5
   |
13 | use tokio::net::UnixStream;
   |     ^^^^^^^^^^^^^^^^^^^^^^ no `UnixStream` in `net`
   |
note: found an item that was configured out
  --> C:\Users\fenix\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tokio-1.21.2\src\net\mod.rs:51:27
   |
51 |     pub use unix::stream::UnixStream;
   |                           ^^^^^^^^^^

For more information about this error, try `rustc --explain E0432`.
error: could not compile `cln-rpc` (lib) due to 2 previous errors
PS D:\dev\workspace\sphinx-swarm\src\bin\cln>

**Comment by fenixlibertas** [comment]
I managed to run on Ubuntu, however there are some errors while starting:

cargo run --bin cln
![image](https://github.com/stakwork/sphinx-swarm/assets/147272698/90f4d637-c699-44aa-a924-940facbe3f30)


and also while starting 
cargo run --bin stack
![image](https://github.com/stakwork/sphinx-swarm/assets/147272698/7f834fe4-0e1e-46e3-9f6d-795aefaa626c)

**Comment by Evanfeenstra** [comment]
@fenixlibertas are you sure you are on latest master branch? 3 days ago I pushed to master to make sure that `cln` error you are seeing would not show up

**Comment by fenixlibertas** [comment]
Yes I am on latest master branch. Errors are visible on images in my previous comment.
Unfortunately, after several attempts and with latest source code I was not able to run this project on Windows and neither on Ubuntu Operating system.
I leave this task to someone else. Perhaps running this on MacOS will go without issues. Thanks.

**Comment by Harshit933** [comment]
Hello @Evanfeenstra . I am willing to take this up. I have set up the backend successfully but I am not able to test the changes that I do to the frontend of the app.
```
cd app 
yarn dev
```
gives me this error :
```
harshit@harshit:~/Desktop/sphinx-swarm$ cd app/ && yarn dev
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'dev'
```

**Comment by Harshit933** [comment]
Okay, I looked into the code more. This is my findings :
- By running `cargo run --lib cln` we are putting up two cln nodes on the regtest network.
- But when we are creating a channel we are checking the on-chain transaction on the testnet as mentioned [here](https://github.com/stakwork/sphinx-swarm/blob/96b961d9e2c137f239189ec6356e684099651c53/app/src/helpers/bitcoin.ts#L1). Thus, the function `getTransactionStatus` always returns false and the `on-chain` transaction confirmation remains 0 resulting in failing to open channel.

**Comment by caydyan** [comment]
Submitted PR: https://github.com/stakwork/sphinx-swarm/pull/716

This is intended for the Sphinx 350,000 sat bounty attached to this issue. BTC payout address: `bc1qev5ant33v5y89qqjvcf4mh9hlax5svqf5xd7gc`.
